src/main/java/com/sublinks/sublinksapi/person/services/LinkPersonCommunityService.java

```diff
@@ -1,109 +1,184 @@
+        .forEach(this.linkPersonCommunityUpdatedPublisher::publish);
+  }
 
-    Collection<Community> communities = new ArrayList<>();
-    for (LinkPersonCommunity linkPersonCommunity : linkPersonCommunities) {
-      communities.add(linkPersonCommunity.getCommunity());
-    }
-    return communities;
+  @Transactional
+  @Override
+  public void deleteLink(LinkPersonCommunity link) {
+
+    this.linkPersonCommunityRepository.delete(link);
+    this.linkPersonCommunityDeletedPublisher.publish(link);
   }
 
-  public Collection<Person> getPersonsFromCommunityAndListTypes(Community community,
-      List<LinkPersonCommunityType> types) {
+  @Transactional
+  @Override
+  public void deleteLink(Community community, Person person,
+      LinkPersonCommunityType linkPersonCommunityType) {
 
-    Collection<LinkPersonCommunity> linkPersonCommunities = linkPersonCommunityRepository.getLinkPersonCommunitiesByCommunityAndLinkTypeIsIn(
-        community, types);
+    final Optional<LinkPersonCommunity> linkOptional = this.getLink(community, person,
+        linkPersonCommunityType);
 
-    return linkPersonCommunities.stream()
-        .map(LinkPersonCommunity::getPerson)
-        .toList();
+    linkOptional.ifPresent((link) -> {
+      this.linkPersonCommunityRepository.delete(link);
+      this.linkPersonCommunityDeletedPublisher.publish(link);
+    });
   }
 
+
+  @Transactional
+  @Override
+  public void deleteLinks(List<LinkPersonCommunity> linkPersonCommunities) {
+
+    this.linkPersonCommunityRepository.deleteAll(linkPersonCommunities);
+
+    linkPersonCommunities.forEach(this.linkPersonCommunityDeletedPublisher::publish);
+
+  }
+
+  @Override
+  public Optional<LinkPersonCommunity> getLink(Community community, Person person,
+      LinkPersonCommunityType linkPersonCommunityType) {
+
+    return this.linkPersonCommunityRepository.getLinkPersonCommunityByCommunityAndPersonAndLinkType(
+        community, person, linkPersonCommunityType);
+  }
+
+  @Override
+  public List<LinkPersonCommunity> getLinks(Person person) {
+
+    return this.linkPersonCommunityRepository.getLinkPersonCommunitiesByPerson(person);
+  }
+
+  @Override
+  public List<LinkPersonCommunity> getLinks(Person person, List<LinkPersonCommunityType> types) {
+
+    return this.linkPersonCommunityRepository.getLinkPersonCommunityByPersonAndLinkTypeIn(person,
+        types);
+  }
+
+  @Override
+  public List<LinkPersonCommunity> getLinks(Person person, LinkPersonCommunityType type) {
+
+    return this.linkPersonCommunityRepository.getLinkPersonCommunityByPersonAndLinkType(person,
+        type);
+  }
+
+  @Override
+  public List<LinkPersonCommunity> getLinksByEntity(Community community, Person person) {
+
+    return this.linkPersonCommunityRepository.getLinkPersonCommunitiesByCommunityAndPerson(
+        community, person);
+  }
+
+  @Override
+  public List<LinkPersonCommunity> getLinksByEntity(Community community,
```

Comment:
Why not just make these both `getLinks` still?

Reply:
![grafik](https://github.com/user-attachments/assets/fa9fc9e5-fdb5-479b-8a06-f357ac4a9e16)
Because then they would have the same "erasure"